### ISSUE TITLE
(module core): how to synthesize template in directory other than temp when running app.synth?

### ISSUE THREAD

**Issue by LazzariPaolo** [aws/aws-cdk#17456]
Going through the quickstart:
https://docs.aws.amazon.com/cdk/latest/guide/hello_world.html

I'm running:

```python
app = core.App()
HelloCdkStack(app, "HelloCdkStack")
s = app.synth()
print(s.directory)
```

the template is always created in a location under `C:\Users\paolo\AppData\Local\Temp`. How can I tell `app.synth` to generate the file in a different directory?
A user should be able to have control over where the template is created

### Proposed Solution

Add a parameter to `app.synth` to allow this

**Comment by LazzariPaolo** [comment]
Apparently the output directory has to be specified at `App` creation time:

`app = core.App(outdir=output_dir)`

weird design choice. What if a user wants to synthesize the template more than once, in two different places?

**Comment by rix0rrr** [comment]
As the documentation for that property says:

> You should never need to set this value. By default, the value you pass to the CLI's --output flag will be used, and if you change it to a different directory the CLI will fail to pick up the generated Cloud Assembly.

https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.AppProps.html#outdir

I'm not quite sure why your template ends up in a temporary directory. It should end up under `cdk.out`. 

In any case, I'll recommend you deploy your stacks using the `cdk deploy` command and not worry about where files end up on your disk.

**Comment by LazzariPaolo** [comment]
@rix0rrr You missed the point of my comment entirely; let me rephrase:

Why does `app.synth()` not offer the option to specify an output directory?

>  I'll recommend you deploy your stacks using the cdk deploy 

I'm not even using `cdk deploy`, I'm just synthesizing templates
